File: cardbattlemodule/src/main/java/com/syahputrareno975/cardbattlemodule/interfaces/ProfileStreamEventController.java
package com.syahputrareno975.cardbattlemodule.interfaces;

import com.syahputrareno975.cardbattlemodule.model.card.CardModel;
import com.syahputrareno975.cardbattlemodule.model.player.PlayerModel;

public interface ProfileStreamEventController {
    void getMyPlayerData(String id);
    void addCardToDeck(PlayerModel p,CardModel c);
    void removeCardFromDeck(PlayerModel p, CardModel c);
    void leaveProfile(SimpleUnit action);
}

File: app/src/main/java/com/syahputrareno975/simplecardbattle/interfaces/ProfileStreamEventController.kt
package com.syahputrareno975.simplecardbattle.interfaces

import com.syahputrareno975.simplecardbattle.model.card.CardModel
import com.syahputrareno975.simplecardbattle.model.player.PlayerModel

interface ProfileStreamEventController {
    fun getMyPlayerData(id : String)
    fun addCardToDeck(p: PlayerModel, c : CardModel)
    fun removeCardFromDeck(p: PlayerModel, c : CardModel)
    fun leaveProfile(action : () -> Unit)
}